File: scala-3/euler/src/main/scala/com/rbleggi/euler/P21.scala
package com.rbleggi.euler

@main def runP21(): Unit = {
  def sumProperDivisors(n: Int): Int = {
    (1 to n / 2).filter(n % _ == 0).sum
  }

  val amicableNumbers = (2 until 10000).filter { a =>
    val b = sumProperDivisors(a)
    b != a && b < 10000 && sumProperDivisors(b) == a
  }

  println(amicableNumbers.sum)
}

File: kotlin/euler/src/main/kotlin/com/rbleggi/euler/P21.kt
package com.rbleggi.euler

fun mainP21() {
    fun sumProperDivisors(n: Int): Int {
        return (1..n / 2).filter { n % it == 0 }.sum()
    }

    val amicableNumbers = (2 until 10000).filter { a ->
        val b = sumProperDivisors(a)
        b != a && b < 10000 && sumProperDivisors(b) == a
    }

    println(amicableNumbers.sum())
}
